specification/redisenterprise/resource-manager/Microsoft.Cache/preview/2022-11-01-preview/redisenterprise.json

```diff
@@ -0,0 +1,1957 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully deleted the database."
+          },
+          "202": {
+            "description": "The database delete operation started. GET the URL in the Azure-AsyncOperation or Location headers to retrieve updated status."
+          },
+          "204": {
+            "description": "Successfully deleted the database."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cache/redisEnterprise/{clusterName}/databases/{databaseName}/listKeys": {
+      "post": {
+        "tags": [
+          "RedisEnterprise",
+          "Databases"
+        ],
+        "operationId": "Databases_ListKeys",
+        "x-ms-examples": {
+          "RedisEnterpriseDatabasesListKeys": {
+            "$ref": "./examples/RedisEnterpriseDatabasesListKeys.json"
+          }
+        },
+        "description": "Retrieves the access keys for the RedisEnterprise database.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ClusterNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DatabaseNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The access keys for the specified database.",
+            "schema": {
+              "$ref": "#/definitions/AccessKeys"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cache/redisEnterprise/{clusterName}/databases/{databaseName}/regenerateKey": {
+      "post": {
+        "tags": [
+          "RedisEnterprise",
+          "Databases"
+        ],
+        "operationId": "Databases_RegenerateKey",
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        },
```

Comment:
We're getting a linter error which seems to suggest we shouldn't do this, and should instead do

          "final-state-via": "location"

Would that work? Do we actually return a valid Location header?

{"pluginName":"spectral","extensionName":"@microsoft.azure/openapi-validator","level":"error","message":"An async POST operation is tracked via Azure-AsyncOperation header. Set 'final-state-via' property to 'location' on 'x-ms-long-running-operation-options'","code":"PostOperationAsyncResponseValidation","details":{"jsonpath":["paths","/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cache/redisEnterprise/{clusterName}/databases/{databaseName}/import","post"],"validationCategory":"RPaaSViolation","providerNamespace":false,"resourceType":false,"range":{"start":{"line":778,"column":15},"end":{"line":834,"column":8}}},"source":[{"document":"[file:///mnt/vss/_work/1/lint-c93b354fd9c14905bb574a8834c4d69b/specification/redisenterprise/resource-manager/Microsoft.Cache/stable/2022-01-01/redisenterprise.json](file:///mnt/vss/_work/1/lint-c93b354fd9c14905bb574a8834c4d69b/specification/redisenterprise/resource-manager/Microsoft.Cache/stable/2022-01-01/redisenterprise.json)","position":{"line":778,"column":7}}]}

Reply:
We're not handling the async case for key regeneration right now. So we could make the change to use `location` here:

`          "final-state-via": "location"`

@riyasimon23

Reply:
@riyasimon23 @jantache-microsoft Yes since the service just always returns 200 today, we should just document that behavior.

Reply:
From discussion with Carl and Riya, we decided to go with the change to `"location"`. This avoids any issues around breaking changes / spec issues, in case we want to change to async operation behaviour in the future (since there is an async op in the background, which just isn't exposed right now). It's also the minimal diff to satisfy the linter for now.